ui/src/app/machines/views/MachineDetails/MachineNetwork/BondForm/utils.test.ts

```diff
@@ -1,32 +1,195 @@
-import { getFirstSelected } from "./utils";
+import { LinkMonitoring } from "./types";
+import { getFirstSelected, getValidNics, preparePayload } from "./utils";
 
+import {
+  BondLacpRate,
+  BondMode,
+  BondXmitHashPolicy,
+} from "app/store/general/types";
+import {
+  NetworkInterfaceTypes,
+  NetworkLinkMode,
+} from "app/store/machine/types";
 import {
   machineDetails as machineDetailsFactory,
   machineInterface as machineInterfaceFactory,
+  networkLink as networkLinkFactory,
 } from "testing/factories";
 
 describe("BondForm utils", () => {
-  it("sorts the nics by name and gets the first interface", () => {
-    const interfaces = [
-      machineInterfaceFactory({
-        name: "bbb",
-      }),
-      machineInterfaceFactory({
-        name: "ccc",
-      }),
-      machineInterfaceFactory({
-        name: "aaa",
-      }),
-    ];
-    const machine = machineDetailsFactory({
-      interfaces,
+  describe("getFirstSelected", () => {
+    it("sorts the nics by name and gets the first interface", () => {
+      const interfaces = [
+        machineInterfaceFactory({
+          name: "bbb",
+        }),
+        machineInterfaceFactory({
+          name: "ccc",
+        }),
+        machineInterfaceFactory({
+          name: "aaa",
+        }),
+      ];
+      const machine = machineDetailsFactory({
+        interfaces,
+      });
+      expect(
+        getFirstSelected(machine, [
+          { nicId: interfaces[0].id },
+          { nicId: interfaces[1].id },
+          { nicId: interfaces[2].id },
+        ])
+      ).toStrictEqual({ nicId: interfaces[2].id });
+    });
+  });
+
+  describe("getValidNics", () => {
+    it("gets interfaces without an existing bond", () => {
```

Comment:
Looks like this test and the next are testing the wrong things

Reply:
Thanks, I've reworded these to make sense.